noir-projects/aztec-nr/aztec/src/note/note_header.nr

```diff
@@ -26,13 +27,13 @@ impl Eq for NoteHeader {
 }
 
 impl NoteHeader {
-    pub fn new(contract_address: AztecAddress, nonce: Field, storage_slot: Field) -> Self {
+    pub fn new(contract_address: AztecAddress, nonce: Field, storage_slot: Point) -> Self {
         NoteHeader { contract_address, nonce, storage_slot, note_hash_counter: 0 }
     }
 }
 
-impl Serialize<4> for NoteHeader {
-    fn serialize(self) -> [Field; 4] {
-        [self.contract_address.to_field(), self.nonce, self.storage_slot, self.note_hash_counter as Field]
+impl Serialize<NOTE_HEADER_LENGTH> for NoteHeader {
+    fn serialize(self) -> [Field; NOTE_HEADER_LENGTH] {
+        [self.contract_address.to_field(), self.nonce, self.storage_slot.x, self.storage_slot.y, self.storage_slot.is_infinite as Field, self.note_hash_counter as Field]
```

Comment:
This increases log sizes and therefore DA, right? It seems related to #7529 and #7515 - perhaps we can prevent x and is infinite from being included here.

Reply:
It does not because when serializing to DA to_be_bytes function is called on the note and not serialize. Clarified that in https://github.com/AztecProtocol/aztec-packages/pull/7437/commits/e77802a4a6e7daee2d82269a82973b7bd393f5e1